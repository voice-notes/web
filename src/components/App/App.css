.App {
  text-align: center;
  background-color: #262626;
  color: white;
  height: 100vh;
}
