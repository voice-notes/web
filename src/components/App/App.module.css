.app {
  background-color: #262626;
  color: white;
  height: 100vh;
}

.btn {
  background-color: black;
  border-radius: 50px 50px 0 0;
  display: flex;
  font-family: var(--primary-font);
  margin: 0 auto;
  padding: 7.5px;
  text-align: center;
  width: 200px;
}

.btnText {
  margin-right: 5px;
  margin-top: 20px;
}
