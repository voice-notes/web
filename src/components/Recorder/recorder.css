.container {
  margin: 0 auto;
  padding: 50px;
  max-width: 950px;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}