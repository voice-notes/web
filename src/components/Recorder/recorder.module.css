.container {
  align-items: center;
  background-color: black;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin: 0 auto;
  max-width: 100%;
  padding: 10px 0;
  text-align: center;
}

.btn {
  font-family: var(--primary-font);
  display: flex;
  margin: 0 auto;
  border-radius: 50px 50px 0 0;
  background-color: black;
  padding: 10px 25px;
  text-align: center;
  
}

.reactIcons {
  color:#BD200C;
  padding: 0;
  margin: 0;
  border: 0;
  font-size: 72px;
}  

.btnTxt {
  margin-top: 24px;
  margin-right: 16px;
}