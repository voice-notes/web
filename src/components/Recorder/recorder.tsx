import React, { useState } from 'react';
import { ReactMic } from 'react-mic';

import { Timer } from '../index';
import styles from './recorder.module.css';
import { RecordingStatus } from '../App/App';

interface Props {
  recordingStatus: RecordingStatus;
  saveBlob: (blob: any) => void;
}

export const Recorder = (props: Props) => {
  const { recordingStatus, saveBlob } = props;

  const [recordingExists, setRecordingExists] = useState<boolean>(false);

  const onData = (recordedBlob: any) => {
    console.log('chunk of real-time data is: ', recordedBlob);
  };

  const onStop = (recordedBlob: any) => {
    saveBlob(recordedBlob);
    setRecordingExists(true);
    console.log('recordedBlob is: ', recordedBlob);
  };

  const isRecording = () => {
    return recordingStatus === 'recording';
  };

  return (
    <div className={styles.container}>
      <ReactMic
        record={isRecording()}
        onStop={onStop}
        onData={onData}
        strokeColor="#0A9E74"
        backgroundColor="#000"
        mimeType="audio/wav"
      />
      <Timer
        recordingStatus={recordingStatus}
        recordingExists={recordingExists}
      />
    </div>
  );
};
