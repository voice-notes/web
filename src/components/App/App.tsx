import React, { useState, useEffect } from 'react';
import { Header } from '../Header/header';
import { Recorder } from '../Recorder/recorder';
import styles from './App.module.css';
import { RecordingIcon } from '../RecordingIcon/recordingIcon';

export const App = () => {
  const [isRecording, setIsRecording] = useState(false); 
  const [buttonText, setButtonText] = useState('Start');

  function toggleRecording() {
    setIsRecording(!isRecording);
  }

  useEffect(() => {
    if(isRecording) {
      setButtonText('Start');
      return;
    }
    setButtonText('Stop');
  })

  return (
    <div className={styles.app}>
      <Header>Taped It</Header>
      <button
        className={styles.btn}
        id="record"
        onClick={() => toggleRecording()}
      >
        <RecordingIcon isRecording={isRecording} />
        <span className={styles.btnTxt}>{buttonText} recording</span>
      </button>
      <Recorder isRecording={isRecording} />
    </div>
  );
};
