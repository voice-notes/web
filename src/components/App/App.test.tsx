import * as React from 'react';
import { render } from '@testing-library/react';
import { shallow, mount } from 'enzyme'
import { App } from './App';
import { Recorder } from '../Recorder/recorder'

jest.mock('react-mic', () => {
  return {
    ReactMic: () => {
      return <div></div>;
    },
  };
});

jest.mock('../Recorder/recorder')

test('renders App', () => {
  const container = render(<App />);
  expect(container).toMatchSnapshot();
});

describe('toggle state', () => {
  it('renders record button', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('button').text()).toContain("<RecordingIcon />Start recording");
  });

  it('updates the button text', async() => {
    const wrapper = mount(<App />);
    wrapper.find('#record').simulate('click')
    await expect(wrapper.find('button').text()).toContain("Stop recording");
  });

  it('toggles back to false', async() => {
    const wrapper = mount(<App />);
    wrapper.find('#record').simulate('click');
    wrapper.find('#record').simulate('click');
    await expect(wrapper.find('button').text()).toContain("Start recording");
  });
})
