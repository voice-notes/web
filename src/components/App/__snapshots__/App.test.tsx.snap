// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        
        <div
          class="container"
        >
          <div
            class="buttonContainer"
          >
            <span
              class="buttonLabelText"
            >
              Record
            </span>
            <button
              class="button"
              id="record"
            >
              <svg
                class="icon"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
              </svg>
            </button>
          </div>
        </div>
        
        <div
          style="display: none;"
        >
          <div
            style="width: 640px; height: 360px;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      
      <div
        class="container"
      >
        <div
          class="buttonContainer"
        >
          <span
            class="buttonLabelText"
          >
            Record
          </span>
          <button
            class="button"
            id="record"
          >
            <svg
              class="icon"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
            </svg>
          </button>
        </div>
      </div>
      
      <div
        style="display: none;"
      >
        <div
          style="width: 640px; height: 360px;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
